src/applications/my-education-benefits/config/form.js

```diff
@@ -780,12 +788,98 @@ const formConfig = {
+                        'view:mailingAddress'
+                      ].address;
+
+                      if (
+                        // formData.showMEBMailingAddressForeign &&
+                        livesOnMilitaryBase &&
+                        country !== 'USA'
+                      ) {
+                        return {
+                          type: 'string',
+                          title: 'APO/FPO',
+                          enum: ['APO', 'FPO'],
+                          enumNames: ['APO', 'FPO'],
+                        };
+                      }
+
+                      return {
+                        type: 'string',
+                        title: 'City',
+                        minLength: 1,
+                        maxLength: 100,
+                        pattern: '^.*\\S.*',
+                      };
+                    },
+                  },
                 },
                 state: {
                   'ui:title': 'State/Province/Region',
                   'ui:errorMessages': {
                     required: 'State is required',
                   },
+                  'ui:options': {
+                    hideIf: formData => {
+                      const { livesOnMilitaryBase } = formData[
+                        'view:mailingAddress'
+                      ];
+                      const { country } = formData[
+                        'view:mailingAddress'
+                      ].address;
+                      return livesOnMilitaryBase && country !== 'USA';
+                    },
+                  },
+                  //   replaceSchema: formData => {
+                  //     const { livesOnMilitaryBase } = formData[
+                  //       'view:mailingAddress'
+                  //     ];
+                  //     const { country } = formData[
+                  //       'view:mailingAddress'
+                  //     ].address;
+                  //
+                  //     if (livesOnMilitaryBase && country !== 'USA') {
+                  //       return {
+                  //         type: 'string',
+                  //         title: 'AE/AA/AP',
+                  //         enum: ['AA', 'AE', 'AP'],
+                  //         enumNames: [
+                  //           'APO/FPO',
+                  //           'APO/FPO (New York)',
+                  //           'APO/FPO (San Francisco)',
+                  //         ],
+                  //       };
+                  //     }
+                  //     return {
+                  //       type: 'string',
+                  //       title: 'State/Province/Region',
+                  //       enum: constants.states.USA.map(state => state.value),
+                  //       enumNames: constants.states.USA.map(
+                  //         state => state.label,
+                  //       ),
+                  //     };
+                  //   },
+                  // },
+                },
+                stateCode: {
```

Comment:
Like I said for the schema, I think a name like `militaryState` would be more descriptive here.